python/pyspark/sql/connect/client/core.py

```diff
@@ -1553,7 +1590,7 @@ def _handle_rpc_error(self, rpc_error: grpc.RpcError) -> NoReturn:
                         info,
                         status.message,
                         self._fetch_enriched_error(info),
-                        True,
+                        self._display_server_stack_trace(),
```

Comment:
Due to GHs abilities to deal with commenting on unmodified lines. I'm wondering if the fix is to simply understand that we're in the error processing code and should not try to go through the usual code process. 

Bascially when we enter handle_rpc_error we set a flag that we're currently processing the error and in `_handle_error` we change the code to

```
def _handle_error(self, error: Exception) -> NoReturn:
    if self._in_error_processing:
       raise error
    # ...
```

This is logically very similar to the recursion guard but more explicit about the fact that when an error happens during error handling we should simply bail out.

Reply:
It's the same logical outcome, more readable and more linear code.

Reply:
Changed to compromise option, with similar structure
```
def _handle_error(self, error: Exception) -> NoReturn:
    if self._in_error_processing:
       raise error
```
but with using class (for more robust, i.e. avoiding information leaking when code excepts)

Reply:
adjusted